aws/rust-runtime/aws-sigv4/benches/sigv4a.rs

```diff
@@ -0,0 +1,31 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+use aws_sigv4::sign::v4a;
+use criterion::{criterion_group, criterion_main, Criterion};
+use std::hint::black_box;
+
+pub fn generate_signing_key(c: &mut Criterion) {
+    c.bench_function("generate_signing_key", |b| {
+        b.iter(|| {
+            black_box({
```

Comment:
I don't think this uses black_box properly. I _think_ you need to blackbox the parameters.

Reply:
I moved it around a bit and re-ran the bench each time. It always stays around 101.33 µs, so I feel like it really is getting tested. If it was being optimized away, it would be much faster, no?